xinference/core/model.py

```diff
@@ -95,20 +97,24 @@ async def _wrap_generator(self, ret: Any):
             return ret
 
     async def generate(self, prompt: str, *args, **kwargs):
-        if not hasattr(self._model, "generate"):
-            raise AttributeError(f"Model {self._model.model_spec} is not for generate.")
-
-        return self._wrap_generator(
-            getattr(self._model, "generate")(prompt, *args, **kwargs)
-        )
+        async with self._lock:
```

Comment:
It seems not work, it returns `IteratorWrapper` and release lock without waiting generate finished.